### PR TITLE
feat(grid-list): #RDV-25 make grid card and show then with list with pagination

### PR DESCRIPTION
**Pull request by niko-amosse**
# PR Description

This PR introduces the necessary components and providers to display availability grids as a list of cards.

## Main Changes

- **Components:**
  - Created modular components for displaying availability grids.
  - Each grid is displayed as a card, with a clear layout optimized for quick viewing.

- **Providers:**
  - Implemented providers to manage the state and data of the availability grids.
  - Providers facilitate the retrieval, processing, and updating of availability data for each card.

## Screenshots

image

## Issue ticket number and link

[RDV-25](https://jira.support-ent.fr/browse/RDV-25)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)
- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/components/CustomMultiAutocomplete/style.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { SxProps } from "@mui/material";
+
+export const chipStyle: SxProps = {
+  margin: "3px",
```
Comment (ThomasSevagen) [inline frontend/src/components/CustomMultiAutocomplete/style.ts:4]: rem

File: frontend/src/components/GridCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+  return (
+    <Box sx={cardWrapperStyle}>
+      <Box sx={leftBoxStyle}>
+        <ColorDot color={grid.color} />
+        <Box sx={leftTextWrapperStyle}>
+          <Box sx={firstLineBoxStyle}>
+            <Typography
+              variant="h5"
+              sx={{
```
Comment (ThomasSevagen) [inline frontend/src/components/GridCard/index.tsx:62]: externaliser

File: frontend/src/containers/MyAvailability/index.tsx [hunk 3]
```diff
@@ -19,26 +31,80 @@ export const MyAvailability: FC = () => {
...
+  );
+
+  const boxRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const updateGridCardSize = () => {
+      if (boxRef.current) {
+        const boxWidth = boxRef.current.offsetWidth;
+        setGridCardSize(
+          boxWidth < 882 ? GRID_CARD_SIZE.SMALL : GRID_CARD_SIZE.LARGE,
```
Comment (FlorentMr) [inline frontend/src/containers/MyAvailability/index.tsx:53]: width en dur

**Review by FlorentMr — COMMENTED** [review]
Trop de px pas assez de rem..

File: frontend/src/core/utils.ts [hunk 4]
```diff
@@ -65,6 +65,10 @@ export const formatPeriodicityToI18n = (periodicity: PERIODICITY): string => {
   }
 };
 
+export const formatDayjsToString = (date: Dayjs): string => {
```
Comment (alicedraillard) [inline frontend/src/core/utils.ts:68]: fichier date.utils.ts dédié

File: frontend/src/containers/GridList/index.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  const handleChangeExpanded = () => {
+    if (gridType === GRID_TYPE.CLOSED && isExpandable)
+      setIsExpanded((prev) => !prev);
+  };
+
+  return (
+    <>
+      {isGridListEmpty ? (
+        <>
+          {gridType === GRID_TYPE.IN_PROGRESS ? (
```
Comment (ThomasSevagen) [inline frontend/src/containers/GridList/index.tsx:50]: &&

File: frontend/src/containers/GridModal/index.tsx [hunk 6]
```diff
@@ -99,32 +110,51 @@ export const GridModal: FC<GridModalProps> = ({ gridModalType }) => {
...
+  };
+
   return (
-    <Modal open={grid}>
-      <Box sx={modalBoxStyle}>
-        <Box sx={contentBoxStyle}>
-          <Box sx={spaceBetweenBoxStyle}>
-            <Typography variant="h3">
-              {t("appointments.create.grid.title")}
+    <>
```
Comment (ThomasSevagen) [inline frontend/src/containers/GridModal/index.tsx:124]: remove

File: frontend/src/containers/MyAvailability/index.tsx [hunk 7]
```diff
@@ -19,26 +31,80 @@ export const MyAvailability: FC = () => {
...
 
   const [gridModalType] = useState<GRID_MODAL_TYPE>(GRID_MODAL_TYPE.CREATION);
+  const [gridCardSize, setGridCardSize] = useState<GRID_CARD_SIZE>(
+    GRID_CARD_SIZE.LARGE,
+  );
+
+  const { gridListLengths } = useAvailabilityProvider();
+  const isAllGridListEmpty = useMemo(
+    () =>
+      gridListLengths[GRID_TYPE.IN_PROGRESS] === 0 &&
```
Comment (ThomasSevagen) [inline frontend/src/containers/MyAvailability/index.tsx:41]: !gridListLengths[GRID_TYPE.IN_PROGRESS]

File: frontend/src/hooks/types.ts [hunk 8]
```diff
@@ -15,13 +15,12 @@ import {
...
   InputsErrors,
   Public,
-  Structure,
 } from "~/providers/GridModalProvider/types";
 
 export interface useUpdateGridInputsReturnType {
   handleNameChange: (e: ChangeEvent<HTMLInputElement>) => void;
   handleColorChange: (color: HexaColor) => void;
-  handleStructureChange: (event: SelectChangeEvent) => void;
+  handleStructureChange: (event: SelectChangeEvent<unknown>) => void;
```
Comment (ThomasSevagen) [inline frontend/src/hooks/types.ts:23]: à checker

File: frontend/src/containers/GridList/index.tsx [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+import { GridCard } from "~/components/GridCard";
+import { GRID_PER_PAGE } from "~/core/constants";
+import { useAvailabilityProvider } from "~/providers/AvailabilityProvider";
+import { GRID_TYPE } from "~/providers/AvailabilityProvider/enum";
+
+export const GridList: FC<GridListProps> = ({ gridType, cardSize }) => {
+  const { t } = useTranslation("appointments");
+  const {
+    gridPages: { [gridType]: page },
+    gridListLengths: {
```
Comment (alicedraillard) [inline frontend/src/containers/GridList/index.tsx:25]: nommage

File: frontend/tsconfig.json [hunk 10]
```diff
@@ -25,7 +25,7 @@
     },
 
     /* Test */
-    "types": ["vitest/globals", "@testing-library/jest-dom"]
+    // "types": ["vitest/globals", "@testing-library/jest-dom"]
```
Comment (alicedraillard) [inline frontend/tsconfig.json:28]: com

**Review by alicedraillard — APPROVED** [review]

**Review by ThomasSevagen — APPROVED** [review]

**Pull request merged**